### DISCUSSION TITLE
UCB Data100: Principles and Techniques of Data Science - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#148] [Announcements]
# UCB Data100: Principles and Techniques of Data Science - CS自学指南

CS自学指南

https://csdiy.wiki/%E6%95%B0%E6%8D%AE%E7%A7%91%E5%AD%A6/Data100/

**Comment by hNSBQZ** [comment]
作业要登陆CalNet Authentication Service是一定本校学生才可以吗

**Reply by parkma99** [reply]
github 上有作业代码，那个是 ucb 的给配好的 jupyter 环境

**Reply by parkma99** [reply]
具体年份从这里找 https://github.com/DS-100

**Reply by hNSBQZ** [reply]
> 具体年份从这里找 https://github.com/DS-100

非常感谢

**Comment by qrdai** [comment]
请问，这课的lab和hw支持otter grader的本地评分嘛？

**Reply by qrdai** [reply]
我看otter官方的文档里面说可以用命令行或python来Non-containerized Grading，但不知道作业里的代码是不是这样支持的

**Reply by 1475505** [reply]
lab支持，hw支持验证答案格式不支持评分

**Reply by 1475505** [reply]
↑ 此回复对应2022sp

**Reply by melodyscallion-a11y** [reply]
> ↑ 此回复对应2022sp

请问前辈能帮忙解释一下怎么样本地验证lab和hw的答案嘛？我找到了otter的documentation 但是里面最开始tutorial文件下载不了 没办法跟着往下走

**Reply by 1475505** [reply]
hw好像没办法，lab春天可以正常安装otter的

**Reply by Cloudmzy** [reply]
所以21fall的是没办法的吗

**Reply by XianglongChen** [reply]
lab春天怎么安装呢～

**Comment by cypppper** [comment]
非常好的课

**Comment by stellalong** [comment]
补充这一门课的先修课要求：
Data 8.
CS 61A or CS88 or Engineering 7. We strongly recommend either CS 61A or CS 88.
EE 16A or Math 54 or Stat 89A. This may be satisfied concurrently with Data 100, but we strongly recommend that you finish a linear algebra course before taking Data 100.

**Reply by wnlz** [reply]
Data8网址：https://www.data8.org/

**Reply by c-chenyi** [reply]
> Data8网址：https://www.data8.org/

Data8视频看不了吧？

**Comment by Cloudmzy** [comment]
同问，如何安装otter包来进行本地作业检查

**Reply by XianglongChen** [reply]
同问～

**Reply by XianglongChen** [reply]
找到了，这里安装otter
https://www.cnblogs.com/tsrigo/p/16653029.html#%E5%AE%89%E8%A3%85otter

**Comment by LiunxRyoJeen** [comment]
非常好， 爱来自中国

**Comment by 2-two-2** [comment]
非常好的课程

**Comment by T1opson** [comment]


**Reply by IndifferentArea** [reply]
广告狗滚吧，好几个课程下面都看到你在叫。咸鱼CalNet的账号999就是你挂的是吧，正常流程一两百就能申请，大家都是学生，做人积点德

**Reply by CodingKirby8620** [reply]
大哥，请问一下，UCB的CalNet账号怎么申请呢？只能说是本校学生申请吗

**Reply by ghost** [reply]
同问calnet账号怎么申请

**Comment by greatxue** [comment]
這門課沒有recording麼？

**Comment by CodingKirby8620** [comment]
诶？这个课程的录像想要观看必须具有权限吗？
还有其他什么地方能看到录像吗

**Reply by Mamihla** [reply]
有的年份是不用权限看recording的，你可以在主页里面找找，比如sp22：https://ds100.org/sp22/

**Reply by cppez** [reply]
最新年份的课好像是不用权限的

**Comment by greatxue** [comment]
上Youtube可以看到Sam Lau的2019版本

**Comment by dghtucs** [comment]
建了一个学习交流群，有意＋643194275

**Comment by p81sunshine** [comment]
grader.check报错：
RuntimeError: Malformed test file: does not define the global variable 'OK_FORMAT'
是因为本地没有下载正确的format吗？
坐标22sp

**Reply by BaiRuic** [reply]
在每个`.py`测试文件的首行添加 `OK_FORMAT=True `就好了

**Comment by ZlatanCN** [comment]
想请教一下各位 ds 的自学路线应该是怎样的😢，本人目前是 22 级数据科学专业的学生，cs61a 暑期内可以完成。

**Reply by Mamihla** [reply]
数据结构61b,数据科学data100，人工智能cs188，概率论进阶eecs126，凸优化eecs127，机器学习入门cs189，深度学习cs182，这个怎么样，全都是伯克利的😀

**Reply by qingzhao0512** [reply]
请问网上都有公开资源吗

**Comment by qua88888** [comment]
请问课程里的homework 怎么访问啊？

**Reply by w-lambda-6** [reply]
github 上有作业代码，那个是 ucb 的给配好的 jupyter 环境，具体年份从这里找 https://github.com/DS-100

**Comment by tj12323** [comment]
在Google云盘的内容怎么找啊

**Comment by wenlingzhi** [comment]
sp2022学习中,贴一个自己写的作业链接, 希望有朋友和我一起互相学习写作业~
https://github.com/wenlingzhi/ds100-sp22

**Comment by Rcpentuy** [comment]
现在能看到的最新版本应该是24spring
视频 https://ds100.org/sp24/
代码 https://github.com/DS-100/sp24-student

**Comment by EvanZhangcn** [comment]
大伙这个环境怎么配的，目前学的是sp24，做到SQL部分作业，环境一直出问题。找官方的发现只有旧教程，且不再更新了，似乎不太行。

**Reply by QXueSheng** [reply]
看一下 lec20 SQLI  lec21 SQLII  里面有pip install 和imdb数据库下载配置安装

**Reply by QXueSheng** [reply]
比如lab10里的 data压缩包直接解压到data文件夹

**Reply by QXueSheng** [reply]
KeyError: '_DEPRECATED_DEFAULT'
The command %config SqlMagic.style = '_DEPRECATED_DEFAULT' is likely to still cause the same error, as the error message indicates that '_DEPRECATED_DEFAULT' is not a valid style option in your current version of prettytable.

Instead, try using a style that actually exists in your version of prettytable:
# First check available styles
import prettytable
print(list(prettytable.TableStyle.__members__.keys()))

# Then use one of the available styles
%config SqlMagic.style = 'DEFAULT'  # or another style from the list

**Reply by c-chenyi** [reply]
这个视频有课程吗？

**Reply by c-chenyi** [reply]
看到视频了，但是2025spring前三个视频被隐藏了

**Comment by bin-shanto** [comment]
可以考虑加上data 8

**Comment by jingningwang** [comment]
这个有后续的进阶课程吗？

**Comment by linyouxue** [comment]
https://github.com/DS-100/fa25-student/blob/main/lecture

**Comment by zhangle1** [comment]
请教下视频教程哪里获取

**Comment by feintenseven** [comment]
这门课程25fall的lab、homework和project部分，除了sql的内容我全部写了参考答案，以及一些课堂笔记，欢迎大家查阅 or 给个star~地址是https://github.com/feintenseven/UCBdata100/tree/main 没有sql是因为不知为何jupyter notebook狂报错。。

**Reply by feintenseven** [reply]
如果后续能解决我会继续更新sql部分的解答！

**Reply by NamingRacksMyBrain** [reply]
hello，请问环境如何配置呢？谢谢

**Reply by feintenseven** [reply]
我马上在我的这个仓库里补充一下！用miniconda比较方便

**Reply by NamingRacksMyBrain** [reply]
看到了！！非常感谢你！

**Reply by NamingRacksMyBrain** [reply]
配置成功：）唯一的小问题是requirement.txt中sympy==1.14.0，但torch==2.5.1强制绑定sympy==1.13.1，把sympy版本强制去除就成功了

**Reply by feintenseven** [reply]
ok！我把这一点修改一下

**Comment by feintenseven** [comment]
今天更新了一下环境的配置方法~可以在我的仓库https://github.com/feintenseven/UCBdata100/tree/main里看到！这门课确实很不错，主要是配套的lab和hw非常赞，还有小猫小狗~

**Reply by feintenseven** [reply]
何意味，markdown自动把我的~变成划线了
